### PR TITLE
Refer to VC-SPECS-DIR for proof types.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address issue #1105 by explicitly stating that `proof` types SHOULD come from the VC-SPECS-DIR. The PR avoids saying "MUST" because VC-SPECS-DIR is not meant to be the authoritative registry of everything VCs, but rather a directory of things that individuals wanted to have listed.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1212.html" title="Last updated on Aug 20, 2023, 5:19 PM UTC (3bcf15a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1212/d84cf2e...3bcf15a.html" title="Last updated on Aug 20, 2023, 5:19 PM UTC (3bcf15a)">Diff</a>

### CODE REVIEW

**Review by OR13 — COMMENTED** [review]
Seems weird to use vc specs dir for data integrity and use the core data model for vc-jwt... I wonder if we can point both to the core data model section, and mention VC specs dir from there.

**Comment by msporny** [comment]
> Seems weird to use vc specs dir for data integrity and use the core data model for vc-jwt... I wonder if we can point both to the core data model section, and mention VC specs dir from there.

I'd be fine with that... this PR is attempting to address an issue you brought up, so you tell me what you want to see in the spec and I'll try to make it happen.

Options include:

* Creating new language in the description for `proof` (this PR)
* Pointing to the Extensibility section in the spec, which talks about VC-JWT and VC-DATA-INTEGRITY and points to VC-SPECS.
* Adding a section on VC-SPECS and talking about the sorts of extensions one could find in there (don't know about this one, feels like we'd just be repeating stuff that's already said in the spec).

Pick a direction to help me refine this PR so that it closes the issue you raised.

**Review by TallTed — COMMENTED** [review]
I'm OK with this. I can also see some justification for tweaking it as @msporny described above. Of the listed options, I'm most inclined to --
> * Add VC-JWT to the section on Proofs in VC-SPECS (or maybe rename that section to "Securing Mechanisms" or something that allows us to put any securing mechanism in there).

**Review by decentralgabe — APPROVED** [review]

**Review by dlongley — APPROVED** [review]

**Review by m00sey — APPROVED** [review]
I like this language:

>Add VC-JWT to the section on Proofs in VC-SPECS (or maybe rename that section to "Securing Mechanisms" or something that allows us to put any securing mechanism in there).

**Review by brentzundel — APPROVED** [review]

File: index.html
Comment (seabass-labrax) [inline index.html:2026]:
```suggestion
<a>verifiable presentation</a> MAY be secured using an <a>external proof</a> such as
```

The use of a hyperlink here seems consistent with earlier usage.

**Review by seabass-labrax — CHANGES_REQUESTED** [review]

**Review by TallTed — APPROVED** [review]

**Comment by msporny** [comment]
@OR13 I need you to signal a direction you'd like this PR to go in among these options https://github.com/w3c/vc-data-model/pull/1212#issuecomment-1646979982 or another one provided by you. Others in the thread have provided input, so if you are aligned w/ their input, we might have a clear path forward with this issue.

**Review by OR13 — CHANGES_REQUESTED** [review]
I think it would be better to link to the specs directory for both securing mechanism... Instead of separating them the way the current text does.

Specifically, we should link to the section of the directory that covers media types that secure the core data model, and I could the core data model media types in that section.

File: index.html
Comment (OR13) [inline index.html:2028]:
```suggestion
mechanism, such as those registered in the
Verifiable Credentials Specifications Directory [[?VC-SPECS]].
```

File: index.html
Comment (jandrieu) [inline index.html:2025]:
```suggestion
Verifiable Credentials Related Specifications Directory [[?VC-SPECS]]. A
```

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-01-vcwg-special#section1-4) on 2023-08-01

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.4. Refer to VC-SPECS-DIR for proof types. (pr vc-data-model#1212)

_See github pull request [vc-data-model#1212](https://github.com/w3c/vc-data-model/pull/1212)._




**Brent Zundel:** this is 'Refer to VC Specs Dir for Proof Types'. two change requests, from Orie and Sebastian.  
… Sebastian, have your changes been addressed?  

**Sebastian Crane:** let me look..  
… yes, those are fine, works for me.  

**Brent Zundel:** manu, looks like Orie has a change suggestion, if you could ..  

**Manu Sporny:** I think I disagree with the change suggestion, mainly because the request was to refer to the VC Specs Dir.  
… not to remove links to the securing specifications we're working on here.  
… my concern is that we would be .. that the question that the PR is trying to address in 1105 is - how should we refer to the securing specs?  
… in other PRs, we said we're going to talk about the securing specs of this WG, we're just going to mention them, and also say there may be other ways to secure the VCs, and those will be in the specs dir.  
… so, the JOSE/COSE stuff, the DI stuff, and for other securing specs, go look in the directory.  
… Orie's change req would be to /not/ mention the securing specs this WG is working on.  
… but only refer to the specs dir, some of which the group has not worked on at all.  

**Joe Andrieu:** I'm concerned about the position you're taking here, Manu.  
… seems to be saying that a method is required to be in a directory.  

**Manu Sporny:** that's not the intent. we're just saying - securing mechanisms exist. Here's two examples. Others are in the specs dir.  

**Joe Andrieu:** I'll take another look.  

**Sebastian Crane:** I think my comment has been addressed.  

**Brent Zundel:** next steps.. we have a change request from Orie. Joe is reviewing..  
… Manu, if at least you could respond to Orie's change req with the things you mentioned today, that'd be valuable.  

**Manu Sporny:** or we could accept it, I don't hear anyone arguing against not mentioning at all.  
… the first objection was "you say there's securing mechanisms, but not providing examples". So we added examples, and that PR went in.  
… and now Orie is saying we should remove that.  
… and if we do that, we'll have no examples again.  
… and won't have a link to the vc specs dir, or external mechanisms.  

**Ted Thibodeau Jr.:** just reading this over, Manu, I think you're mistaken as to what it currently says.  
… it says, if present, proof value should be .. [ quote follows ].  
… now, it's only a SHOULD, it does allow for whatever private or public deployment.  
… but it does say that you SHOULD use one that's in the directory.  

**Manu Sporny:** ah, I see the reading you're referring to.  
… also don't see VC-JWT or VC-JOSE/COSE.  

**Brent Zundel:** sounds like there's a bit of tweaking needed.  

**Manu Sporny:** yes.  

**Brent Zundel:** sufficient steps taken, lets move on to 1215.  

</details>

File: index.html
Comment (msporny) [reply index.html:2028]: If we take this change request, @OR13 -- none of the VC-COSE-JOSE securing mechanisms will be listed due to:

https://github.com/w3c/vc-specs-dir/pull/14 being blocked.

Only the VC Data Integrity mechanisms will be listed. 

Given the opposition we've traditionally had to favoring one mechanism over the other (your change request would favor Data Integrity over VC-COSE-JOSE) I suggest we don't take this change now until https://github.com/w3c/vc-specs-dir/pull/14 is unblocked, or we're listing all securing mechanism in the vc-specs-dir.

Comment (OR13) [reply index.html:2028]: @Sakurann ^ seems you are block the dependency, and I am blocking this PR, based on your block.. can we resolve?

Comment (OR13) [reply index.html:2028]: @msporny I hear what you are saying, but lets unblock the stuck PR .... by closing it, or merging it... before we take this change... its been open since April 8th... it should have been closed or merged by now.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-09-vcwg#section2-5) on 2023-08-09

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.5. Refer to VC-SPECS-DIR for proof types. (pr vc-data-model#1212)

_See github pull request [vc-data-model#1212](https://github.com/w3c/vc-data-model/pull/1212)._




**Manu Sporny:** PR 1212 examples of securing mechanisms in spec. Point to specifications or directory? Need PR about media types?  

**Orie Steele:** VCs with some securing mechanisms, with DI proofs; two specs; or media types; This or that language in DM spec.  
… merge media types, refer to them consistently.  

**Joe Andrieu:** this establishes related specs into a privileged position...  

> *Orie Steele:* +1 on should to MAY.

> *Manu Sporny:* +1 on SHOULD to MAY conversion.

> *Ivan Herman:* +1 to MAY.

**Manu Sporny:** securing mechanisms we have vetted here and those not. Anyone can add to specs dir. No review...  

> *Michael Jones:* I want XML DSIG.

**Manu Sporny:** very dangerous thing; any mechanism...  

> *Orie Steele:* +1 JoeAndrieu.

**Joe Andrieu:** I think anything does go; people can come up with new crypto; a directory is okay; our mechanisms are published as recs.  

**Sebastian Crane:** be careful, don't devalue our (WG) opinion.  

**Kristina Yasuda:** safely change to MAY...  

**Manu Sporny:** we don't say what has/hasn't been vetted in registry? The VC DM doesn't say what has been vetted.  
… how should we refer to securing mechanisms we have been working on?  

> *Orie Steele:* I suggested a concrete change here: [https://github.com/w3c/vc-data-model/pull/1212/files#r1279836059](https://github.com/w3c/vc-data-model/pull/1212/files#r1279836059).

**Manu Sporny:** what sections/where to put?  

**Orie Steele:** if media types is merge is will be obvious;.  

_See github pull request [vc-specs-dir#14](https://github.com/w3c/vc-specs-dir/pull/14)._




> *Orie Steele:* +1 manu.

**Manu Sporny:** blocking on Kristina PR 14; create media type in specs dir; then merge.  

> *Orie Steele:* I can edit "register to list".

> *Gabe Cohen:* directister.

**Joe Andrieu:** avoid that directory is a registry.  

> *Ted Thibodeau Jr.:* Sorry, I'm slow, I have to re-review most recent changes in [https://github.com/w3c/vc-specs-dir/pull/14](https://github.com/w3c/vc-specs-dir/pull/14).

_See github issue [vc-specs-dir#27](https://github.com/w3c/vc-specs-dir/issues/27)._




</details>

File: index.html
Comment (Sakurann) [reply index.html:2028]: https://github.com/w3c/vc-specs-dir/pull/14 is unblocked.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-15-vcwg-special#section1-4) on 2023-08-15

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.4. Refer to VC-SPECS-DIR for proof types. (pr vc-data-model#1212)

_See github pull request [vc-data-model#1212](https://github.com/w3c/vc-data-model/pull/1212)._




**Brent Zundel:** next up PR 1212: Refer to VC-SPECS-DIR for proof types. many approvals. one outstanding request from Orie who is not on the call.  

**Manu Sporny:** looking to see where the objection is...would rather link to the specs dir for both securing mechanisms for this particular PR re:vc-specs-dir#14 which was merged.  
… we could refer to two parts in the specs dir now. that feels awkward. can just point to two specifications that the group is working on. can make those changes if no objections.  

**Brent Zundel:** Orie has joined, which changes would you like to see in 1212. sorry for ambushing you.  

**Orie Steele:** expect there to be other representations of VCs. expect there to be media types that distinguish them. easier to refer to the specs dir where those media types exist, instead of our work item continuing to refer to the two current mechanisms over and over again.  
… we have a specs dir and should be using it to clarify that it's not just Data Integrity Proofs or JWTs. you'll need a media type to tell them apart.  

**Manu Sporny:** I will refer to the specs directory and two different places: one for proofs, one for media types.  

**Orie Steele:** I would refer to just the media types section, and add to the section vc+ld+json as a media type which can contain embedded proofs. just refer to the media type section.  

**Manu Sporny:** previously kristina had objected putting the base media type into the media types section. if you're OK with that we can do what Orie said.  
… will add a PR to the specs dir for that media type and then in this PR point to that location.  

> *Phillip Long:* +1 to manu's course of action.

**Brent Zundel:** any objections? [none heard].  

</details>

**Comment by msporny** [comment]
Editorial, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**
